internal/extensions/resolve.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2023-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensions
+
+import (
+	"buf.build/gen/go/bufbuild/protovalidate/protocolbuffers/go/buf/validate"
+	"google.golang.org/protobuf/encoding/protowire"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/descriptorpb"
+	"google.golang.org/protobuf/types/dynamicpb"
+)
+
+const (
+	legacyExtensionIndex protowire.Number = 51071 // protovalidate versions < v0.2.0
+)
+
+//nolint:gochecknoglobals // static data, only want single instance
+var resolver = newExtensionResolver()
+
+// Resolve resolves extensions without using [proto.GetExtension], in case the
+// underlying type of the extension is not the concrete type expected by the
+// library. In some cases, particularly when using a dynamic descriptor set, the
+// underlying extension value's type will be a dynamicpb.Message. In some cases,
+// the extension may not be resolved at all. This function handles reparsing the
+// fields as needed to get it into the right concrete message as needed. Resolve
```

Comment:
```suggestion
// fields as needed to get it into the right concrete message. Resolve
```

Reply:
Fixed in 9d5c4696e205f7a84d0d8ba131dd42d1ea2193fe